src/interactivity/win32/uiaTextRange.cpp

```diff
@@ -10,12 +10,15 @@
 
 using namespace Microsoft::Console::Types;
 using namespace Microsoft::Console::Interactivity::Win32;
+using namespace Microsoft::WRL;
 using Microsoft::Console::Interactivity::ServiceLocator;
 
-std::deque<UiaTextRange*> UiaTextRange::GetSelectionRanges(_In_ IUiaData* pData,
-                                                           _In_ IRawElementProviderSimple* pProvider)
+HRESULT UiaTextRange::GetSelectionRanges(_In_ IUiaData* pData,
+                                         _In_ IRawElementProviderSimple* pProvider,
+                                         _Out_ std::deque<ComPtr<UiaTextRange>>& ranges)
+try
 {
-    std::deque<UiaTextRange*> ranges;
+    typename std::remove_reference<decltype(ranges)>::type temporaryResult;

```

Comment:
what in the heck is this doing

Reply:
![image](https://user-images.githubusercontent.com/11050425/68701582-f469b880-053b-11ea-904a-710fc68a737f.png)

Reply:
I think with multiple people being confused on this, we should probably leave a comment on how it works or what it's there for. Especially since I'm still not exactly sure how it works. Even if the comment is just "this is magic to get the type of the `termporaryResult` right"